lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
> It's not just that. Even if we cache the results for filesystem calls, we still have to monitor the paths to respond to filesystem changes.

That's a great point and I agree it's both unfortunate and less than ideal. What I am worried about is inconsistency, though: for example, what should we pass from the `tabs` package? After a quick glance, it doesn't seem to have any reference to a `File` object, and therefore would make the icon service report an incorrect information. Maybe we could let `tabs` conform to the same interfaces as `tree-view` (i.e. by querying the filesystem), but wouldn't that lead to the same performance considerations we're making for the `file-icons` package?

I think having a better abstraction in Atom core (where we provide a central facility that collects information about the filesystem and watches files as they change over time) would be the optimal scenario, but in the meantime it seems like the only alternative to return a correct result that is also consistent across calls is to bake fs queries into the file icons service package. What do you think? 💭